rmw_zenoh_cpp/src/detail/rmw_context_impl_s.cpp

```diff
@@ -417,37 +439,6 @@ class rmw_context_impl_s::Data final
   std::unordered_map<const rmw_node_t *, std::shared_ptr<rmw_zenoh_cpp::NodeData>> nodes_;
 };
 
-///=============================================================================
-static void graph_sub_data_handler(z_loaned_sample_t * sample, void * data)
```

Comment:
For this initial port, it would be best if we retained this function while updating the API. We can rely on a lambda or std::bind specify this function as the callback.

Reply:
In this particular case, I'm fine with doing this as a lambda; the code in here is simple enough.  We will eventually need to go through and remove the `data_to_data_shared_ptr` structure in favor of lambda captures, but that will be a separate PR.

Reply:
We decided to stick with the lambdas here, so resolving this.